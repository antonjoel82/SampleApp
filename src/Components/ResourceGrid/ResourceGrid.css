.res-grid {
	text-align: center;
	display: grid;
	grid-gap: 1rem;
	grid-template-columns: repeat(auto-fill, minmax(325px, 1fr));
	grid-template-rows: span 1;
	padding: 1rem 3rem 1rem 3rem;
}