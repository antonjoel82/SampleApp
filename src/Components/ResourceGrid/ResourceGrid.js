import React from 'react';
import Box from '@material-ui/core/Box';
import GridResourceCard from '../GridResourceCard/GridResourceCard.js';

import { Grid } from '@material-ui/core';

const ResourceGrid = ({ resources }) => {
    const resArr = resources || [];

    return (
      <Box px={10} py={3}>
        <Grid container spacing={2}>
          {
            resArr.map((resource, i) => {
              return (
                <Grid key={i /*resource.id*/} item>
                  <GridResourceCard key={i /*resource.id*/} resource={resource}/>
                </Grid>
              );
            })
          }
        </Grid>
      </Box>
    );
};
export default ResourceGrid;
