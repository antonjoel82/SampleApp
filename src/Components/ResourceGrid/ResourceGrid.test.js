import React from 'react';
import { shallow } from 'enzyme';

import ResourceGrid from './ResourceGrid';

describe('ResourceGrid', () => {
  it('should render correctly in "debug" mode', () => {
    const comp = shallow(<ResourceGrid resources={[]} debug />);
    expect(comp).toMatchSnapshot();
  });
  it('should work if no props are provided', () => {
    const comp = shallow(<ResourceGrid />);
    expect(comp).toMatchSnapshot();
  });
});
