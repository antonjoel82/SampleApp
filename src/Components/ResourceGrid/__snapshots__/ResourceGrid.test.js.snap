// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceGrid should render correctly in "debug" mode 1`] = `
<Styled(MuiBox)
  px={10}
  py={3}
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  />
</Styled(MuiBox)>
`;
